src/main/java/org/jenkinsci/plugins/edithelp/EditHelpRootAction.java

```
@@ -0,0 +1,149 @@
+    }
+
+    //save text area into file and array
+    public void doUpdateHelpInfo() {
+        //process get request
+        String className = Stapler.getCurrentRequest().getParameter("class");
+        String updatedClassText = Stapler.getCurrentRequest().getParameter("textArea");
+        //replacement or creation of the string
+        arrayOfHelp.put(className+".html",updatedClassText);
+
+        //writing into a file
+        if (className != null) {
+            Jenkins jn = Jenkins.getInstance();
+            if (jn != null) {
+                File dirfile = jn.getRootDir();         
+                String dirName = null;         
+                if(dirfile != null) {
+                    //check if the directory exists
+                    File helpManagerFile = new File(dirfile, "helpmanager");
+                    if(!helpManagerFile.exists())
+                        if (!helpManagerFile.mkdirs()) {
+                            LOGGER.log(FINE,"unable create directory");
+                        }
+
+                    File newFile = new File(helpManagerFile.getAbsolutePath() + "/" + className + ".html");
+                    //check if the file exists
+                    if (!newFile.exists()) {
+                        try{
+                            if (!newFile.createNewFile()) {
+                                LOGGER.log(FINE,"unable create directory");
+                            }
+                        }
+                        catch (IOException exname){}
+
+                    }
+
+                    //stream writing into file
+                    try{
+                        BufferedWriter out = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(newFile), StandardCharsets.UTF_8));
+                        out.write(updatedClassText);
+                        out.flush();
+                        out.close();
+                    }
+                    catch (IOException exname){}
```

empty catch block is very bad - log or rethrow exception